### DISCUSSION TITLE
rabbitmqctl status command causes the CPU to surge high

### DISCUSSION THREAD

**Discussion by polaris-alioth** [rabbitmq/rabbitmq-server#3864] [Other] [answered]
When we use `rabbitmqctl status` to do health check in the livenessProbe, somehow it leads the CPU to surge high, then the server down and container restart. It seems that  `rabbitmqctl status` is not a good choose to do health check in k8s env. (our livenessProbe periodSeconds is 10s)
I wonder why `rabbitmqctl status` leads the CPU to surge high and if there is an alternative solution to replace  `rabbitmqctl status` in the livenessProbe.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
If you wander about something, please start a discussion instead of filing an issue. The issue template on GitHub strongly suggests that.

**Comment by michaelklishin** [comment, marked as answer]
`rabbitmq-diagnostics status` collects and returns quite a bit of information. `rabbitmq-diagnostics report` does all that plus metrics of virtually every live entity in the system, so, that's obviously extremely expensive.

There are [several much smaller, much more focused](https://www.rabbitmq.com/monitoring.html#health-checks) health checks available these days in a dedicated doc section, as well as a note on [what monitoring frequency is recommended](https://www.rabbitmq.com/monitoring.html#monitoring-frequency).

However, without any data provided, if `rabbitmq-diagnostics status` causes node CPU usage to go above its quota, the quota might be way too low. So make sure the node has at least two CPU cores. RabbitMQ was not designed for single core or fraction-of-a-core  environments.
